### PR TITLE
feat: bubble up minion response events

### PR DESCRIPTION
**Pull request by rfrm**

### CODE REVIEW

**Comment by DiegoRBaquero** [comment]
In which scenario do we want this? one can set per minion response listener, why have it in the army level?

**Comment by rfrm** [comment]
@DiegoRBaquero I'll use this to make tests a little bit cleaner. The way it has been done is to get the minion from the Army using the event key like `internals.worker = worker.minions['event-name']`, and then listen for the response event on that minion. I want to make it more encapsulated as you won't need to go deep into the army's minions array.

File: lib/index.js [hunk 1]
```diff
@@ -90,6 +90,7 @@ module.exports = (_manifest) => {
         minion.on('ready', checkReady);
         minion.on('message', (m, meta) => eventEmitter.emit('message', worker.config.name, m, meta));
         minion.on('error', (e) => eventEmitter.emit('error', e));
+        minion.on('response', (response) => eventEmitter.emit('response', { minion, response }));
```
Comment (ryanhallcs) [inline lib/index.js:93]: why return the entire minion object in the response?

Comment (rfrm) [reply lib/index.js]: @ryanhallcs yeah, updated to return just the response.

**Pull request merged**